test/Plugin/HalTest.php

```diff
@@ -1010,12 +1010,16 @@ public function testRenderingPaginatorCollectionRendersPaginationAttributes()
             'page_size',
             'total_items',
             'page',
+            'start_index',
+            'end_index',
         ];
         $this->assertEquals($expected, array_keys($rendered));
         $this->assertEquals(100, $rendered['total_items']);
         $this->assertEquals(3, $rendered['page']);
         $this->assertEquals(10, $rendered['page_count']);
         $this->assertEquals(10, $rendered['page_size']);
+        $this->assertEquals(20, $rendered['start_index']);
+        $this->assertEquals(29, $rendered['end_index']);
```

Comment:
While this correctly ensures that the newly introduced attributes are present, we also need tests covering what happens in terms of which items are in the collection when those attributes are present in the initial request.

Ideally, all of these, including the above assertions, should be in a separate tests to ensure that _existing_ behavior remains unchanged, while the _new_ behavior is also present.

I'd have tests demonstrating:

- what happens when both items are present in the request
- what happens when only the `start_index` is present in the request (for instance, how does this interact with pagination?)
- what happens when only the `end_index` is present in the request (is this valid? if so, how does it interact with pagination?)